File: Nordeus-Challenge/Assets/_Scripts/Models/Combat/CombatResponse.cs
using System;
using System.Collections.Generic;

[Serializable]
public class CombatResponse
{
    public string combatId;
    public List<CombatEvent> events;
    public CombatState state;
    public PlayerStateResponse playerState;
    public string currentTurn;    // "PLAYER" or "ENEMY" while active; null when combat ended
    public string environmentId;  // which environment this combat takes place in
}

File: backend/src/main/java/com/example/backend/dtos/responces/CombatResponse.java
package com.example.backend.dtos.responces;

import com.example.backend.models.combats.CombatEvent;
import com.example.backend.models.combats.CombatState;

import java.util.List;
import java.util.UUID;

public record CombatResponse(UUID combatId, List<CombatEvent> events, CombatState state, PlayerStateResponse playerState, String currentTurn, String environmentId) {}
